### PR TITLE
[SR-6671] Decimal.leastNonzeroMagnitude, leastNormalMagnitude should have -128 exponent

### PR DESCRIPTION
**Pull request by takasek**
Fixes https://bugs.swift.org/browse/SR-6671 .

### CODE REVIEW

**Comment by takasek** [comment]
@swift-ci please test

**Comment by parkera** [comment]
This one is shared with the overlay so we should make coordinated changes there if we go forward with it.

**Comment by phausler** [comment]
Can we add some tests for this to verify the expected behavior?

**Comment by takasek** [comment]
> we should make coordinated changes there

Do you mean that these codes should be fixed at the same time?

https://github.com/apple/swift/blob/49a8cf050304e3503960f0f1815afafe71dba28a/test/stdlib/TestDecimal.swift#L130-L133
https://github.com/apple/swift/blob/8f8f00012a391b7c254e2056ef382c768ca7671a/stdlib/public/SDK/Foundation/Decimal.swift#L22-L23

> Can we add some tests for this to verify the expected behavior?

Sure, please add them.

**Comment by parkera** [comment]
@takasek I think @phausler was asking for you to write some additional tests and include them in this PR.

**Comment by takasek** [comment]
OK, I'll add some tests.

File: TestFoundation/TestJSONSerialization.swift [hunk 1]
```diff
@@ -1257,8 +1257,8 @@ extension TestJSONSerialization {
...
         XCTAssertEqual(try trySerialize([Decimal.leastFiniteMagnitude]), "[-3402823669209384634633746074317682114550000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000]")
-        XCTAssertEqual(try trySerialize([-Decimal.leastNonzeroMagnitude]), "[-0.0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001]")
-        XCTAssertEqual(try trySerialize([Decimal.leastNonzeroMagnitude]), "[0.0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001]")
+        XCTAssertEqual(try trySerialize([-Decimal.leastNonzeroMagnitude]), "[-0.00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001]")
+        XCTAssertEqual(try trySerialize([Decimal.leastNonzeroMagnitude]), "[0.00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001]")
```
Comment (takasek) [inline TestFoundation/TestJSONSerialization.swift]: These tests failed because of changes in this PR.

File: Foundation/Decimal.swift [hunk 2]
```diff
@@ -101,31 +101,31 @@ public struct Decimal {
 
 extension Decimal {
     public static let leastFiniteMagnitude = Decimal(
-        _exponent: 127,
+        _exponent: Int32(Int8.max),
```
Comment (takasek) [inline Foundation/Decimal.swift]: What do you think of such expression change?

**Comment by millenomi** [comment]
@swift-ci Please test.

**Comment by alblue** [comment]
@phausler @parkera can you check this over - are the added tests sufficient?

File: Foundation/Decimal.swift (hunk 2 above)
Comment (stephentyrone) [reply Foundation/Decimal.swift]: What's going on here, and where did this property come from? "Magnitudes" are always non-negative, but this is a huge negative number. I don't know why this property even exists, but the name and documentation are *extremely* misleading.

(@takasek This is tangential to your change, but this is the first I've seen this).

**Comment by millenomi** [comment]
Are the objections above understood? Do they still need correction?

**Comment by alblue** [comment]
This seems to have gone stale. Is it still needed, or can it be closed?

**Comment by spevans** [comment]
@swift-ci test

**Comment by takasek** [comment]
Any problem still on this issue?

**Comment by alblue** [comment]
@swift-ci test

**Comment by takasek** [comment]
This issue is yet stale.
`NSDecimalNumber` defines its exponent's range is down to -128.
https://developer.apple.com/documentation/foundation/nsdecimalnumber
Thus `Foundation.Decimal` should follow it.

**Comment by shahmishal** [comment]
The Swift project moved the default branch to `main` and deleted `master` branch, so GitHub automatically closed the PR. Please re-create pull request with main branch.

More detail about the branch update - https://forums.swift.org/t/updating-branch-names/40412

**Pull request closed without merging**